### PR TITLE
added ingress support section

### PR DESCRIPTION
**Pull request by bfallonf**
For: #8596 

cc: @JacobTanenbaum @zhaozhanqi

### CODE REVIEW

**Comment by bfallonf** [comment]
Neato. Thanks @JacobTanenbaum 

@openshift/team-documentation PTAL

File: architecture/networking/routes.adoc [hunk 1]
```diff
@@ -833,3 +833,61 @@ $ oc expose service/<name> --hostname="m.api.openshift.org"
...
+
+This controller watches ingress objects and creates one or more routes to
+satisfy the conditions of the ingress object. The controller is also responsible
+for keeping the ingress object and generated route objects synchronized. This
+includes giving generated routes permissions on the secrets associated with the
+ingress object. 
+
+For example, an ingress object configured as:
+
+----
```
Comment (gaurav-nelson) [inline architecture/networking/routes.adoc]: maybe add `[source, yaml]` tags before code blocs (2 instances)

File: architecture/networking/routes.adoc [hunk 2]
```diff
@@ -833,3 +833,61 @@ $ oc expose service/<name> --hostname="m.api.openshift.org"
...
+  - host: test.com
+    http:
+     paths:
+     - path: /test
+       backend:
+        serviceName: test-1
+        servicePort: 80
+----
+
+would generate the following route object:
```
Comment (gaurav-nelson) [inline architecture/networking/routes.adoc]: s/would generate/generates

File: architecture/networking/routes.adoc [hunk 3]
```diff
@@ -833,3 +833,61 @@ $ oc expose service/<name> --hostname="m.api.openshift.org"
...
+  to:
+    name: test-1
+  port:
+     targetPort: 80
+----
+<1> The name is generated by the route objects, with the ingress name as a prefix.
+
+[NOTE]
+====
+In order for a route to be created, an ingress object needs to have a host,
```
Comment (gaurav-nelson) [inline architecture/networking/routes.adoc]: s/In order for/For

Comment (gaurav-nelson) [reply architecture/networking/routes.adoc]: maybe? s/needs to have/must have

**Review by gaurav-nelson — CHANGES_REQUESTED** [review]
few nits, rest looks good. 👍

**Review by gaurav-nelson — APPROVED** [review]

**Comment by bfallonf** [comment]
Thanks @gaurav-nelson . I made changes as per your suggestions.

**Comment by vikram-redhat** [comment]
@bfallonf - was this QEed?

**Comment by bfallonf** [comment]
@vikram-redhat yes, in the PR this was based on.

**Comment by vikram-redhat** [comment]
Removing dedicated and online labels as it doesn't apply to them.

**Pull request merged**